// f1-api/dist/parsers/driver.js
Object.defineProperty(exports, "__esModule", { value: true });
const parser_1 = require("./parser");
class DriverParser extends parser_1.default {
    static getInstance() {
        if (!this.instance) {
            this.instance = new DriverParser();
        }
        return this.instance;
    }
    parse(data) {
        let driverData = data;
        if (data === null || data === void 0 ? void 0 : data.Drivers) {
            if (!(data === null || data === void 0 ? void 0 : data.Drivers.length)) {
                return null;
            }
            driverData = data.Drivers[0];
        }
        const { driverId, permanentNumber, code, givenName, familyName, dateOfBirth, nationality } = driverData;
        const driver = {
            id: driverId,
            code,
            number: parseInt(permanentNumber, 10),
            firstName: givenName,
            lastName: familyName,
            birthday: dateOfBirth,
            nationality,
        };
        return driver;
    }
}
exports.default = DriverParser;
