// f1-api/dist/parsers/driverStanding.js
Object.defineProperty(exports, "__esModule", { value: true });
const parser_1 = require("./parser");
const driver_1 = require("./driver");
const constructors_1 = require("./constructors");
class DriverStandingParser extends parser_1.default {
    static getInstance() {
        if (!this.instance) {
            this.instance = new DriverStandingParser();
        }
        return this.instance;
    }
    parse(data) {
        const constructorsParser = constructors_1.default.getInstance();
        const driverParser = driver_1.default.getInstance();
        const { position, points, wins, Driver: driverData, Constructors: constructors } = data;
        const driverStanding = {
            position: parseInt(position, 10),
            points: parseFloat(points),
            wins: parseInt(wins, 10),
            driver: driverParser.parse(driverData),
            constructors: constructorsParser.parse({
                Constructors: [...constructors],
            }),
        };
        return driverStanding;
    }
}
exports.default = DriverStandingParser;
