// f1-api/dist/parsers/raceSchedule.js
Object.defineProperty(exports, "__esModule", { value: true });
const parser_1 = require("./parser");
const circuit_1 = require("./circuit");
class RaceScheduleParser extends parser_1.default {
    static getInstance() {
        if (!this.instance) {
            this.instance = new RaceScheduleParser();
        }
        return this.instance;
    }
    parse(data) {
        const circuitParser = circuit_1.default.getInstance();
        let raceScheduleData = data;
        if (data === null || data === void 0 ? void 0 : data.Races) {
            if (!(data === null || data === void 0 ? void 0 : data.Races.length)) {
                return null;
            }
            raceScheduleData = data.Races[0];
        }
        const { season, round, raceName: name, Circuit: circuit, date, time } = raceScheduleData;
        const raceSchedule = {
            season: parseInt(season, 10),
            round: parseInt(round, 10),
            name,
            circuit: circuitParser.parse(circuit),
            date: new Date(`${date}T${time}`),
        };
        return raceSchedule;
    }
}
exports.default = RaceScheduleParser;
